portal/src/components/leaderboard/LeaderboardAdminView.tsx

```diff
@@ -0,0 +1,315 @@
+                  return (
+                    <Flex
+                      key={leaderboard.id}
+                      p="10px"
+                      rounded="lg"
+                      bgColor={index == selectedIndex ? "#4D4D4D" : "auto"}
+                      onClick={() => {
+                        setSelectedIndex(index);
+                      }}
+                    >
+                      <Text>{leaderboard.title}</Text>
+                    </Flex>
+                  );
+                })}
+              </Box>
+            </Box>
+          </Box>
+          <Spacer />
+          <Box>
+            <Button
+              w="100%"
+              h="40px"
+              bgColor="#FFFFFF"
+              color="#232323"
+              leftIcon={<AddIcon w="8px" h="8px" />}
+              onClick={() => {
+                setStatus("create");
+              }}
+            >
+              Add new leaderboard
+            </Button>
+          </Box>
+        </Flex>
+        <Flex
+          w="800px"
+          minH="720px"
+          bgColor={panelBackground}
+          m="20px"
+          p="20px"
+          rounded="lg"
+          flexDir="column"
+        >
+          {selectedLeaderboard.data && (
+            <>
+              {status == "normal" && (
+                <>
+                  <LeaderboardMetadata
+                    leaderboard={selectedLeaderboard.data}
+                    lastUpdate={lastUpdate}
+                    handleEdit={() => {
+                      setStatus("edit");
+                    }}
+                  />
+                  <LeaderboardUpload
+                    handleScores={async (scores: Score[]) => {
+                      return await pushLeaderboardScores(selectedLeaderboard.data.id, scores);
+                    }}
+                  />
+                  <Button
+                    my="20px"
+                    bgColor="#FFFFFF"
+                    color="#232323"
+                    alignSelf="center"
+                    onClick={onOpen}
+                  >
+                    Show leaderboard API
+                  </Button>
+                  <Modal isOpen={isOpen} onClose={onClose}>
+                    <ModalOverlay />
+                    <ModalContent
+                      bg="#2D2D2D"
+                      border="1px solid white"
+                      borderRadius="20px"
+                      textColor="white"
+                      mx="15px"
+                      maxW="500px"
+                    >
+                      <ModalHeader fontSize="18px" fontWeight="900">
+                        Leaderboard API
+                      </ModalHeader>
+                      <ModalBody>
+                        <LeaderboardAPI id={selectedLeaderboard.data.id} />
+                      </ModalBody>
+                    </ModalContent>
+                  </Modal>
+                </>
+              )}
+              {status == "create" && (
```

Comment:
It's inside {selectedLeaderboard.data && (
so it doesn't render when list is empty